### PR TITLE
absolute position stacks relative to the parent

### PR DESCRIPTION
**Pull request by christhesoul**
Enables the absolute positioning of element (using `top:` and `left:`) relative to its parent.

We do this by setting the every element to relative positioning unless either `top` or `left` are specified, and thus the assumption is the programmer wishes to absolutely position this element relative to its parent.

For clarity, an absolutely positioned element needs to be inside an element that has relative or absolute positioning itself.

_plus_ fun flags to prove it all works. :-)

### CODE REVIEW

File: examples/flags/mauritius.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require "scarpe"
+
+Scarpe.app(title: "Mauritius", height: 300) do
```
Comment (andrewn617) [inline examples/flags/mauritius.rb:3]: 🇲🇺  !

**Review by andrewn617 — APPROVED** [review]

File: lib/scarpe/flow.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ class Flow < Scarpe::Widget
     include Scarpe::Background
     include Scarpe::Border
 
-    def initialize(width: nil, height: nil, margin: nil, margin_left: nil, margin_top: nil, &block)
+    def initialize(width: nil, height: nil, top: nil, margin: nil, margin_left: nil, margin_top: nil, &block)
```
Comment (chitty) [inline lib/scarpe/flow.rb:8]: It looks like we're missing ` @top = top`

File: lib/scarpe/document_root.rb [hunk 3]
```diff
@@ -30,7 +30,7 @@ def do_js_eval(js)
     end
 
     def empty
-      "<body id='body-#{html_id}'><div id='wrapper-#{html_id}'></div></body>"
+      "<body id='body-#{html_id}' style='border:0;margin:0;'><div id='wrapper-#{html_id}'></div></body>"
```
Comment (noahgibbs) [inline lib/scarpe/document_root.rb:33]: If this is merged before mine, I'll need to update for this (and the height-adjust in app.rb). If mine is merged first, feel free to grab me and I can do the merge or help you do so.

**Review by noahgibbs — APPROVED** [review]

**Review by Schwad — APPROVED** [review]
If you need a hand with merge conflicts give us a headsup in the discord!

**Comment by Schwad** [comment]
@christhesoul I wonder if this needs to be revisited with the latest implementation 🤔

**Comment by noahgibbs** [comment]
Nick: at a minimum it'll need to move into the wv files and Webview classes. I don't think they existed when this was written.

**Comment by Schwad** [comment]
For tracking purposes I'm marking this and similar issues as "possibly abandoned"

**Comment by Schwad** [comment]
let me know if you want to revisit this

**Pull request closed without merging**